generators/E906LegacyGen/SQPrimaryVertexGen.C

```diff
@@ -16,34 +16,39 @@ from Kun to E1039 experiment in Fun4All framework
 
 #include "SQPrimaryVertexGen.h"
 //#include "SQBeamlineObject.h"
-
-SQPrimaryVertexGen::SQPrimaryVertexGen() 
+double beam_vtx_x; 
+double beam_vtx_y;
+ TF2* beam_global;
+SQPrimaryVertexGen::SQPrimaryVertexGen():
+ _beam_profile(nullptr)
 {
 inited = false;
-
-}
-SQPrimaryVertexGen::~SQPrimaryVertexGen(){}
-
-void SQPrimaryVertexGen::Initfile(){
+ // beam_vtx_x = 0;
+ // beam_vtx_y = 0;
  
 }
 
-void SQPrimaryVertexGen::InitRun(PHCompositeNode* topNode)
+SQPrimaryVertexGen::~SQPrimaryVertexGen()
 {
-  beam_profile= false;
-  beamProfile = NULL;
-  beamProfile = new TF2("beamProfile", "exp(-0.5*(x-[0])*(x-[0])/[1]/[1])*exp(-0.5*(y-[2])*(y-[2])/[3]/[3])", -10., 10., -10., 10.);
-  beamProfile->SetParameter(0, 0.0);
-  beamProfile->SetParameter(1, 0.68);
-  beamProfile->SetParameter(2, 0.0);
-  beamProfile->SetParameter(3, 0.76);
-
-  // beamProfile = new TF2("beamProfile", "exp(-0.5*(x-0.)*(x-0.)/0.414/0.414)*exp(-0.5*(y-0.)*(y-0.)/0.343/0.343)", -10., 10., -10., 10.);
+	delete _beam_profile;
```

Comment:
if (_beam_profile) delete _beam_profile;

Reply:
I will implement that..thank you